### PR TITLE
quincy: test/pybind: replace nose with pytest

### PR DESCRIPTION
**Pull request by cbodley**
backport tracker: https://tracker.ceph.com/issues/63937

---

backport of https://github.com/ceph/ceph/pull/52143
parent tracker: https://tracker.ceph.com/issues/61567

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by cbodley** [comment]
there were several conflicts around filtering for the crimson-rados suite; hopefully i got everything right

**Review by ljflores — APPROVED** [review]

File: qa/suites/crimson-rados/basic/tasks/rados_python.yaml
Comment (idryomov) [inline qa/suites/crimson-rados/basic/tasks/rados_python.yaml:19]: bogus `wait or`

File: qa/suites/crimson-rados/rbd/tasks/rbd_python_api_tests_old_format.yaml
Comment (idryomov) [inline qa/suites/crimson-rados/rbd/tasks/rbd_python_api_tests_old_format.yaml:14]: bogus `-m 'not skip_if_crimson'`

File: qa/workunits/rbd/test_librbd_python.sh
Comment (idryomov) [inline qa/workunits/rbd/test_librbd_python.sh:8]: bogus `"$@"` here and below

File: src/test/pybind/test_rbd.py
Comment (idryomov) [inline src/test/pybind/test_rbd.py:792]: bogus mark -- `SKIP_IF_CRIMSON` that was used for nose isn't in quincy

File: src/test/pybind/test_rbd.py
Comment (idryomov) [inline src/test/pybind/test_rbd.py:2017]: ditto

File: src/test/pybind/test_rados.py
Comment (idryomov) [inline src/test/pybind/test_rados.py:408]: bogus mark?

File: src/test/pybind/test_rados.py
Comment (idryomov) [inline src/test/pybind/test_rados.py:946]: bogus mark, also below -- I don't see `rollback` and `wait` in quincy at all?

File: src/test/pybind/test_rados.py
Comment (idryomov) [inline src/test/pybind/test_rados.py:1281]: ditto

File: src/test/pybind/pytest.ini
Comment (idryomov) [inline src/test/pybind/pytest.ini:6]: Looks like `rollback`, `skip_if_crimson` and `wait` markers could go.

**Review by idryomov — COMMENTED** [review]
> there were several conflicts around filtering for the crimson-rados suite; hopefully i got everything right

I spotted some leftovers, mostly around markers.  Otherwise LGTM!

**Comment by ljflores** [comment]
@cbodley let me know when this is ready to get into testing

**Comment by cbodley** [comment]
thanks @idryomov, cleaned up the cherry-picks and documented the conflicts around markers

File: qa/suites/crimson-rados/basic/tasks/rados_python.yaml
Comment (cbodley) [inline qa/suites/crimson-rados/basic/tasks/rados_python.yaml:19]: oops, missed an `or` here

**Comment by idryomov** [comment]
jenkins test make check

**Comment by idryomov** [comment]
jenkins test api

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
@cbodley Backport looks great, but I noticed that `wait` marker used in RADOS tests is missing from `pytest.ini` in main.  Is that intentional?

**Comment by cbodley** [comment]
> @cbodley Backport looks great, but I noticed that `wait` marker used in RADOS tests is missing from `pytest.ini` in main. Is that intentional?

not intentional, no

**Comment by ljflores** [comment]
Waiting on this PR to go into testing / merge to unblock https://trello.com/c/Yjrx9ygD/1911-wip-yuri8-testing-2023-12-15-0911

**Comment by cbodley** [comment]
seeing one failure from test_rbd.py, ex. http://qa-proxy.ceph.com/teuthology/yuriw-2024-01-09_20:56:52-rbd-quincy-distro-default-smithi/7510134/teuthology.log
```
=================================== FAILURES ===================================
_________________________ TestImage.test_diff_iterate __________________________

self = <test_rbd.TestImage object at 0x7fa827636dd8>

    def test_diff_iterate(self):
        check_diff(self.image, 0, IMG_SIZE, None, [])
        self.image.write(b'a' * 256, 0)
        check_diff(self.image, 0, IMG_SIZE, None, [(0, 256, True)])
        self.image.write(b'b' * 256, 256)
        check_diff(self.image, 0, IMG_SIZE, None, [(0, 512, True)])
        self.image.discard(128, 256)
        check_diff(self.image, 0, IMG_SIZE, None, [(0, 512, True)])

        self.image.create_snap('snap1')
        self.image.discard(0, 1 << IMG_ORDER)
        self.image.create_snap('snap2')
        self.image.set_snap('snap2')
>       check_diff(self.image, 0, IMG_SIZE, 'snap1', [(0, 512, False)])

../../../clone.client.0/src/test/pybind/test_rbd.py:1226:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../clone.client.0/src/test/pybind/test_rbd.py:1450: in check_diff
    eq(extents, expected)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = [], b = [(0, 512, False)]

    def assert_equal(a, b):
>       assert a == b
E       AssertionError

../../../clone.client.0/src/test/pybind/assertions.py:2: AssertionError
```
@idryomov any idea what might be happening there?

**Comment by idryomov** [comment]
@cbodley See https://tracker.ceph.com/issues/63941.   I have updated pacific branch in ceph-ci.git, it's possible that other named branched need to be pushed as well.

**Comment by idryomov** [comment]
@cbodley Scratch the above, I see that it's a different test.  The root cause is the same binaries vs test mismatch.

I think you just need to rebase to pick up https://github.com/ceph/ceph/pull/54950 that landed in quincy in the interim.

**Comment by ljflores** [comment]
jenkins test api

**Pull request merged**
